internal/wrphandlers/mocktr181/handler.go

```diff
@@ -205,32 +205,70 @@ func (h Handler) set(tr181 *Tr181Payload) (int64, []byte, error) {
 		Names:      tr181.Names,
 		StatusCode: http.StatusAccepted,
 	}
+
+	var (
+		writableParams []*MockParameter
+		failedParams   []Parameter
+	)
+	// Check for any parameters that are not writable.
 	for _, parameter := range tr181.Parameters {
+		var found bool
 		for i := range h.parameters {
 			mockParameter := &h.parameters[i]
 			if mockParameter.Name != parameter.Name {
 				continue
 			}
 
-			switch mockParameter.Access {
-			case "w", "wr", "rw":
-				mockParameter.Value = parameter.Value
-				mockParameter.DataType = parameter.DataType
-				mockParameter.Attributes = parameter.Attributes
-				result.Parameters = append(result.Parameters, Parameter{
-					Name:       mockParameter.Name,
-					Value:      mockParameter.Value,
-					DataType:   mockParameter.DataType,
-					Attributes: mockParameter.Attributes,
-					Message:    "Success",
-				})
-			default:
-				result.Parameters = append(result.Parameters, Parameter{
-					Name:    mockParameter.Name,
-					Message: "Parameter is not writable",
-				})
-				result.StatusCode = 520
+			// Check whether mockParameter is writable.
+			if strings.Contains(mockParameter.Access, "w") {
+				found = true
+				// Add mockParameter to the list of parameters to be updated.
+				writableParams = append(writableParams, mockParameter)
+				continue
```

Comment:
```suggestion
				// Move onto the next requested parameter.
				break
```

@schmidtw can we assume there aren't any duplicates?